### PR TITLE
Misc style improvements

### PR DESCRIPTION
**Pull request by gregestren**
- Upgrades / downgrades some section header sizes to clearer positions
- Mild reorganization for clearer content flow
- Capitalization typo fixes

Serves https://github.com/bazelbuild/bazel/issues/10794

### CODE REVIEW

File: site/docs/updating-bazel.md [hunk 1]
```diff
@@ -5,11 +5,21 @@ title: Updating Bazel
 
 # Updating Bazel
 
-Bazel defined a [backwards compatibility policy](https://docs.bazel.build/versions/master/backward-compatibility.html)
+Bazel has a [backwards compatibility policy](https://docs.bazel.build/versions/master/backward-compatibility.html)
```
Comment (aiuto) [inline site/docs/updating-bazel.md]: Question to think about.   Should we use 'Bazel' to mean bazel the code or in this broader sense of 'the bazel project'.
Or should we have distinct words to disambiguate.

File: site/docs/updating-bazel.md [hunk 2]
```diff
@@ -5,11 +5,21 @@ title: Updating Bazel
...
 (see [guidance for rolling out incompatible changes](https://www.bazel.build/breaking-changes-guide.html) if you are the author of one).
-This page summarized best practices on how to test and migrate your project with upcoming incompatible
+This page summarizes best practices on how to test and migrate your project with upcoming incompatible
```
Comment (aiuto) [inline site/docs/updating-bazel.md]: _That_ page?

File: site/docs/updating-bazel.md [hunk 3]
```diff
@@ -5,11 +5,21 @@ title: Updating Bazel
...
 changes and how to provide feedback to the incompatible change authors.
 
+## Managing Bazel versions with Bazelisk
+
+Bazel team implemented a Bazel wrapper called [bazelisk](https://github.com/bazelbuild/bazelisk) that helps you
```
Comment (aiuto) [inline site/docs/updating-bazel.md]: Do we normally use  |bazel team| or |the bazel team|

File: site/docs/updating-bazel.md (hunk 1 above)
Comment (jin) [reply site/docs/updating-bazel.md]: +1 "The Bazel maintainers have a.."

Comment (gregestren) [reply site/docs/updating-bazel.md]: Done.

File: site/docs/updating-bazel.md (hunk 3 above)
Comment (gregestren) [reply site/docs/updating-bazel.md]: Done.

File: site/docs/updating-bazel.md (hunk 2 above)
Comment (gregestren) [reply site/docs/updating-bazel.md]: Done.

**Pull request closed without merging**